### PR TITLE
Remove references to TLS-SNI-01

### LINKED ISSUE #6445: [documentation] TLS-SNI-01 is deprecated
Documentation's pages mentioning TLS-SNI-01 should indicate the deprecation status.

- https://github.com/certbot/certbot/blob/master/docs/challenges.rst
- https://github.com/certbot/certbot/blob/master/docs/using.rst

API documentation should be updated too.

https://community.letsencrypt.org/t/february-13-2019-end-of-life-for-all-tls-sni-01-validation-support/74209

> February 13, 2019: End-of-Life for All TLS-SNI-01 Validation Support

### PR DESCRIPTION
**Pull request by schoen**
Building on parts of #6470 and superseding it.

Fixes #6445.

### CODE REVIEW

File: docs/challenges.rst [hunk 1]
```diff
@@ -16,21 +15,21 @@ design favors performing challenges automatically, and this is the normal case f
...
-* Apache plugin: (TLS-SNI-01) Tries to edit your Apache configuration files to temporarily serve 
-  a Certbot-generated certificate for a specified name. Use the Apache plugin when you're running 
-  Certbot on a web server with Apache listening on port 443.
-* NGINX plugin: (TLS-SNI-01) Tries to edit your NGINX configuration files to temporarily serve a 
-  Certbot-generated certificate for a specified name. Use the NGINX plugin when you're running 
-  Certbot on a web server with NGINX listening on port 443.
+* Apache plugin: (HTTP-01) Tries to edit your Apache configuration files to temporarily serve files to
+  satisfy challenges from the certificate authority. Use the Apache plugin when you're running Certbot on a
+  web server with Apache listening on port 80.
+* nginx plugin: (HTTP-01) Tries to edit your nginx configuration files to temporarily serve files to
```
Comment (adferrand) [inline docs/challenges.rst]: Leading capital letter for Nginx, like for the others sections in this list?

File: docs/contributing.rst [hunk 2]
```diff
@@ -187,7 +187,7 @@ Authenticators
 
 Authenticators are plugins that prove control of a domain name by solving a
 challenge provided by the ACME server. ACME currently defines three types of
-challenges: HTTP, TLS-SNI, and DNS, represented by classes in `acme.challenges`.
+challenges: HTTP, TLS-SNI (deprecated), TLS-ALPR, and DNS, represented by classes in `acme.challenges`.
```
Comment (adferrand) [inline docs/contributing.rst]: Four types of challenge, if we let the TLS-SNI be in the enumeration here. Or we remove it completely and forget about it? Technically this challenge is still in the spec and the servers that implement the spec, but not for long.

Comment (schoen) [reply docs/contributing.rst]: Oops, thanks @adferrand!

File: docs/challenges.rst [hunk 3]
```diff
@@ -16,21 +15,21 @@ design favors performing challenges automatically, and this is the normal case f
...
-* Standalone plugin: (TLS-SNI-01 or HTTP-01) Tries to run a temporary web server listening on either HTTP on 
-  port 80 (for HTTP-01) or HTTPS on port 443 (for TLS-SNI-01). Use the Standalone plugin if no existing program 
-  is listening to these ports. Choose TLS-SNI-01 or HTTP-01 using the `--preferred-challenges` option.
+* Standalone plugin: (HTTP-01) Tries to run a temporary web server listening on HTTP on port 80. Use the
+  Standalone plugin if no existing program is listening to this port.
 * Manual plugin: (DNS-01 or HTTP-01) Either tells you what changes to make to your configuration or updates 
   your DNS records using an external script (for DNS-01) or your webroot (for HTTP-01). Use the Manual 
-  plugin if you have the technical knowledge to make configuration changes yourself when asked to do so. 
+  plugin if you have the technical knowledge to make configuration changes yourself when asked to do so,
+  and are prepared to repeat these steps every time the certificate needs to be renewed.
```
[the remark this answers is not in the document]
Comment (schoen) [reply docs/challenges.rst]: That's what I meant for this change to address (!). So maybe I need to remove it or rephrase it.

File: docs/using.rst [hunk 4]
```diff
@@ -44,32 +44,31 @@ a combination_ of distinct authenticator and installer plugins.
...
+manual_     Y    N    | Helps you obtain a certificate by giving you instructions to  http-01_ (80) or
+                      | perform domain validation yourself. Additionally allows you   dns-01_ (53) 
+                      | to specify scripts to automate the validation task in a 
                       | customized way.
 =========== ==== ==== =============================================================== =============================
 
 .. |dns_plugs| replace:: :ref:`DNS plugins <dns_plugins>`
 
 Under the hood, plugins use one of several ACME protocol challenges_ to
 prove you control a domain. The options are http-01_ (which uses port 80),
```
[the remark this answers is not in the document]
Comment (schoen) [reply docs/using.rst]: I support the proper use of commas!

**Pull request merged**